hiku/validate/query.py

```diff
@@ -198,18 +199,28 @@ def visit_root(self, obj):
 
 class _RecordFieldsValidator(QueryVisitor):
 
-    def __init__(self, field_names, errors):
-        self._field_names = set(field_names)
+    def __init__(self, fields, errors):
+        self._fields = fields
         self._errors = errors
 
     def visit_field(self, obj):
-        if obj.name not in self._field_names:
+        if obj.name not in self._fields:
             self._errors.report('Unknown field name')
         elif obj.options is not None:
             self._errors.report('Options are not expected')
+        elif _AssumeRecord().visit(self._fields[obj.name]):
```

Comment:
I think that `self._fields` variable should have more descriptive name, as this mapping contains field types